### ISSUE TITLE
screen flickering

### ISSUE THREAD

**Issue by kpcyrd** [ParadoxSpiral/axon#2]
It seems the whole screen is redrawn every time something changes on screen, this is causing flickering on xfce4-terminal because you can very briefly see the blank screen while it's being redrawn. Maybe this can be optimized somehow, for example by only redrawing lines that changed, or maybe there's a more efficient way to do this in curses itself.

**Comment by ParadoxSpiral** [comment]
Yes, this is a fairly serious problem. I've been able to ignore it because the terminal emulator I use (`alacritty`) is fast or smart enough to properly render.

The nice solution would be damage tracking, that is only rendering widgets that actually changed like you said. I think curses might have this built in, but I didn't like the look of curses and decided to roll my own, which is essentially a tree of widgets that is rendered each time a render is requested.
